tern/analyze/docker/dockerfile.py

```diff
@@ -265,3 +266,38 @@ def get_base_image_tag(base_instructions):
     if len(image_tag_list) == 1:
         image_tag_list.append('')
     return tuple(image_tag_list)
+
+
+def find_git_info(line, dockerfile_path):
+    '''Given a line of ADD command and the path of dockerfile,
+    return the information(string format) on the git project name and sha
+    if the <src> is a git repository or included in a git repository.
+    ADD command has a general format:
+    ADD [--chown=<user>:<group>] <src> <dst>
+    '''
+    args = line.split(' ')
+    src_path = ''
+    # check if --chown exists
+    if args[1].startswith('--chown'):
+        # check if the line is valid
+        if len(args) < 4:
+            logger.error('Invalid ADD command line')
+        src_path = args[2]
+    else:
+        # the line has no --chown option
+        if len(args) < 3:
+            logger.error('Invalid ADD command line')
+        src_path = args[1]
+    # get the absolute src_path
+    comment_line = common.check_git_src(dockerfile_path, src_path)
+    return comment_line
+
+
+def expand_add_command(dfobj):
+    dockerfile_path = dfobj.filepath
+    for i, command_dict in enumerate(dfobj.structure):
+        if command_dict['instruction'] == 'ADD':
+            comment_line = find_git_info(command_dict['content'],
+                                         dockerfile_path)
+            dfobj.structure[i]['content'] = command_dict['content'].strip('\n')\
+                + ' # ' + comment_line + '\n'
```

Comment:
You are updating the `dfobj.structure[i]['content']` correctly here, but we also need to update `dfobj.structure[i]['value']` in the structure dictionary to match. Note that the `value` is the same as the associated `content` value minus the Dockerfile command and newline character. Example of the dictionary from a run command:

```{'instruction': 'RUN', 'startline': 12, 'endline': 12, 'content': 'RUN pip3 install --upgrade pip && pip3 install tern\n', 'value': 'pip3 install --upgrade pip && pip3 install tern'}```